File: server/Models/Events/Event.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models.Event;
public class Event
{
  public int Id { get; set; }
  public Guid Uid { get; set; } = Guid.NewGuid();
  public int ClientId { get; set; }
  public Guid ClientUid { get; set; } = Guid.NewGuid();
  public string ClientFirstName { get; set; } = "";
  public string ClientLastName { get; set; } = "";
  public string? BusinessName { get; set; } = "";
  public string ClientPhone { get; set; } = "";
  public string ClientEmail { get; set; } = "";
  public string? EventName { get; set; }
  public DateTime EventStart { get; set; }
  public DateTime EventEnd { get; set; }
  public Guid? BillingAddressEntryUid { get; set; } = Guid.NewGuid();
  public string BillingFirstName { get; set; } = "";
  public string BillingLastName { get; set; } = "";
  public string BillingAddressLine1 { get; set; } = "";
  public string? BillingAddressLine2 { get; set; }
  public string BillingCity { get; set; } = "";
  public string BillingState { get; set; } = "";
  public string BillingZipCode { get; set; } = "";
  public string BillingPhone { get; set; } = "";
  public string BillingEmail { get; set; } = "";
  public Guid? DeliveryAddressEntryUid { get; set; } = Guid.NewGuid();
  public string DeliveryFirstName { get; set; } = "";
  public string DeliveryLastName { get; set; } = "";
  public string DeliveryAddressLine1 { get; set; } = "";
  public string? DeliveryAddressLine2 { get; set; }
  public string DeliveryCity { get; set; } = "";
  public string DeliveryState { get; set; } = "";
  public string DeliveryZipCode { get; set; } = "";
  public string DeliveryPhone { get; set; } = "";
  public string DeliveryEmail { get; set; } = "";
  public EventStatus Status { get; set; } = EventStatus.Draft;
  public string EventType { get; set; } = "";
  public string? Notes { get; set; }
  public string? InternalNotes { get; set; }
  public List<LogisticsTask> LogisticsTasks { get; set; } = [];
  public List<EventItem> Items { get; set; } = [];
  public decimal Subtotal { get; set; }
  public decimal TaxAmount { get; set; }
  public List<Discount> Discounts { get; set; } = [];
  public decimal Total { get; set; }
  public List<Payment> Payments { get; set; } = [];
  public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
  public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum EventStatus
{
  Draft,
  Confirmed,
  Scheduled,
  Completed,
  Cancelled
}

File: client/src/containers/Events/types/Event.ts
import type { ItemBasics } from "../CreateEvent/CreateEvent";
import type { PaymentMethod } from "./PaymentType";

type Event = {
  uid: string;
  clientUid: string;
  clientFirstName: string;
  clientLastName: string;
  businessName: string | null;
  clientPhone: string;
  clientEmail: string;
  eventName: string | null;
  eventStart: string;
  eventEnd: string;
  billingAddressEntryUid: string;
  billingFirstName: string;
  billingLastName: string;
  billingAddressLine1: string;
  billingAddressLine2: string | null;
  billingCity: string;
  billingState: string;
  billingZipCode: string;
  billingPhone: string;
  billingEmail: string;
  deliveryAddressEntryUid: string;
  deliveryFirstName: string;
  deliveryLastName: string;
  deliveryAddressLine1: string;
  deliveryAddressLine2: string | null;
  deliveryCity: string;
  deliveryState: string;
  deliveryZipCode: string;
  deliveryPhone: string;
  deliveryEmail: string;
  status: EventStatus;
  notes: string | null;
  eventType: string;
  internalNotes: string | null;
  logisticsTrips: LogisticsTrip[];
  items: EventItem[];
  subtotal: number;
  taxAmount: number;
  discounts: EventDiscount[];
  total: number;
  transactions: Transaction[];
  createdAt: string;
  updatedAt: string;
};

type EventStatus =
  | "Draft"
  | "Confirmed"
  | "Scheduled"
  | "On Hold"
  | "Completed"
  | "Cancelled";

type ItemType = "AlaCarte" | "Package";

type DiscountType = "FlatAmount" | "Percentage";

type EventItem = {
  uid: string;
  inventoryItemUid: string;
  description: string;
  unitPrice: number;
  type: string;
  packageUid: string;
  quantity: number;
  inventoryItemSKU: string;
  createdAt: string;
  updatedAt: string;
};

export type LogisticsTrip = {
  uid: string;
  actualArrival: string;
  actualStart: string;
  completedAt: string;
  deliveryDetails: LogisticsTripDeliveryDetails;
  crew: CrewMember[];
  crewLeadName: string;
  scheduledEnd: string;
  scheduledStart: string;
  status: string;
  tripSummary: string;
  truckUid: string;
  truckName: string;
  workItems: WorkItem[];
};

type LogisticsTripDeliveryDetails = {
  addressLine1: string;
  addressLine2?: string;
  city: string;
  state: string;
  zipCode: string;
  phone: string;
  eventName: string;
  eventUid: string;
};

type CrewMember = {
  fullName: string;
  isLead: boolean;
  roleNotes: string;
  uid: string;
  userUid: string;
};

type WorkItem = {
  completedAt: Date;
  isCompleted: boolean;
  notes: string;
  type: string;
  uid: string;
};

type TransactionType = "Payment" | "Refund" | "Adjustment";

export type Transaction = {
  uid: string;
  eventUid: number;
  amount: number;
  type: TransactionType;
  method: PaymentMethod;
  occurredAt: string;
  externalTransactionId: string | null;
  relatedTransactionUid: string | null;
  processedBy: string | null;
  notes: string | null;
  cardBrand: string;
  last4: string;
};

type EventDiscount = {
  uid: string;
  eventId: number;
  type: DiscountType;
  amount: number;
  reason: string | null;
  createdAt: string;
};

type EventDraft = {
  clientUid: string;
  eventName: string;
  deliveryDate: Date;
  deliveryTime: string;
  pickupDate: Date;
  pickupTime: string;
  billingAddressUid: string;
  deliveryAddressUid: string;
  notes: string;
  internalNotes: string;
  items: ItemBasics[];
};

type EventDraftResponse = {
  uid: string;
  status: string;
  clientName: string;
  total: number;
};

export type {
  Event,
  EventDiscount,
  EventDraft,
  EventItem,
  EventDraftResponse,
  EventStatus,
  WorkItem,
  CrewMember,
};
